### PR TITLE
simplewallet: new "address one-off <major> <minor>" command

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

File: src/wallet/wallet2.cpp [hunk 1]
```diff
@@ -1548,6 +1548,15 @@ void wallet2::expand_subaddresses(const cryptonote::subaddress_index& index)
   }
 }
 //----------------------------------------------------------------------------------------------------
+void wallet2::create_one_off_subaddress(const cryptonote::subaddress_index& index)
+{
+  const crypto::public_key pkey = get_subaddress_spend_public_key(index);
+  std::vector<std::pair<crypto::public_key, cryptonote::subaddress_index>> new_elements;
+  new_elements.push_back(std::make_pair(pkey, index));
+  boost::container::ordered_unique_range_t tag;
+  m_subaddresses.insert(tag, new_elements.begin(), new_elements.end());
```
Comment (selsta) [inline src/wallet/wallet2.cpp]: This does not compile.

File: src/simplewallet/simplewallet.cpp [hunk 2]
```diff
@@ -194,7 +194,7 @@ namespace
                             "  account tag <tag_name> <account_index_1> [<account_index_2> ...]\n"
                             "  account untag <account_index_1> [<account_index_2> ...]\n"
                             "  account tag_description <tag_name> <description>");
-  const char* USAGE_ADDRESS("address [ new <label text with white spaces allowed> | all | <index_min> [<index_max>] | label <index> <label text with white spaces allowed> | device [<index>]]");
+  const char* USAGE_ADDRESS("address [ new <label text with white spaces allowed> | all | <index_min> [<index_max>] | label <index> <label text with white spaces allowed> | device [<index>] [one-off <major> <minor>]]");
```
Comment (selsta) [inline src/simplewallet/simplewallet.cpp:197]:
```suggestion
  const char* USAGE_ADDRESS("address [ new <label text with white spaces allowed> | all | <index_min> [<index_max>] | label <index> <label text with white spaces allowed> | device [<index>] | one-off <major> <minor>]");
```

**Comment by moneromooo-monero** [comment]
You mean s/minor/account/ and s/major/subaddress/ in the help ?

**Comment by selsta** [comment]
Yes

File: src/simplewallet/simplewallet.cpp [hunk 3]
```diff
@@ -9282,6 +9282,25 @@ bool simple_wallet::print_address(const std::vector<std::string> &args/* = std::
...
+      return true;
+    }
+    uint32_t major, minor;
+    if (!epee::string_tools::get_xtype_from_string(major, local_args[0]) || !epee::string_tools::get_xtype_from_string(minor, local_args[1]))
+    {
+      fail_msg_writer() << tr("failed to parse index: ") << local_args[0] << " " << local_args[1];
+      return true;
+    }
+    m_wallet->create_one_off_subaddress({major, minor});
+    print_address_sub(m_wallet->get_num_subaddresses(m_current_subaddress_account) - 1);
```
Comment (selsta) [inline src/simplewallet/simplewallet.cpp:9301]: Why does this get printed here?

**Pull request merged**